include/seastar/core/shared_ptr.hh

```diff
@@ -290,17 +290,31 @@ public:
     lw_shared_ptr(std::nullptr_t) noexcept : lw_shared_ptr() {}
     lw_shared_ptr(const lw_shared_ptr& x) noexcept : _p(x._p) {
         if (_p) {
+#pragma GCC diagnostic push
+        // to silence the false alarm from GCC 12, see
+        // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105204
+#if defined(__GNUC__) && (__GNUC__ >= 12)
```

Comment:
https://godbolt.org/z/dqY4xrTKT

Reply:
ahh, thanks for pointing this out. forgot that clang "pretended" to be GCC!